zulipterminal/cli/run.py

```diff
@@ -294,6 +309,21 @@ def list_themes() -> str:
     )
 
 
+def print_welcome_message(zterm: Dict[str, Any]) -> None:
+    print("Loading with:")
+    print(WELCOME_MESSAGE_TEXT['theme'](zterm['theme']))
+    complete, incomplete = complete_and_incomplete_themes()
+    if zterm['theme'][0] in incomplete:
+        print(in_color('yellow',
+                       "   WARNING: Incomplete theme; "
+                       "results may vary!\n"
+                       "      (you could try: {})".
+                       format(", ".join(complete))))
+    zterm.pop('theme')
+    for key in sorted(zterm):
+        print(WELCOME_MESSAGE_TEXT[key](zterm[key]))
```

Comment:
This is pretty fancy, but the standard line follows something like:
```
"   autohide setting '{}' specified {}.".format(*autohide),
```
On that basis, I think we could massage the second part of zterm to loop through everything, and for theme perhaps have "extra text" which is otherwise `None`?
(all this in a third commit, based on my main feedback)

Reply:
I updated the pr to make it simpler. Yeah, looking back I also think this was really fancy and could have been simply avoided.